backend/models/schemas.py

```
@@ -0,0 +1,97 @@
+    birth_time: str
+    timezone: str
+    latitude: float
+    longitude: float
+    
+    @validator('birth_date')
+    def validate_birth_date(cls, v):
+        try:
+            datetime.strptime(v, '%Y-%m-%d')
+            return v
+        except ValueError:
+            raise ValueError('Birth date must be in YYYY-MM-DD format')
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)